src/hotspot/cpu/s390/templateInterpreterGenerator_s390.cpp

```diff
@@ -1180,7 +1180,10 @@ void TemplateInterpreterGenerator::generate_fixed_frame(bool native_call) {
   __ z_stg(Z_R0, _z_ijava_state_neg(monitors), fp);
 
   __ add2reg(Z_esp, -Interpreter::stackElementSize);
-  __ z_stg(Z_esp, _z_ijava_state_neg(esp), fp);
+
+  __ z_sgrk(Z_R0, Z_esp, fp);
+  __ z_srag(Z_R0, Z_R0, Interpreter::logStackElementSize);
+  __ z_stg(Z_R0, _z_ijava_state_neg(esp), fp);
```

Comment:
Please use `save_esp()` here to avoid code duplication.

Reply:
`save_esp()` uses `Z_fp` which is `Z_R9`. But here `fp` refers to `Z_R11` or `Z_tmp_2`. 

One possibility is to modify `save_esp` and make sure that it takes `fp` as an argument. Other way it leave it as-it-is. Which one do you prefer.

Reply:
Hmm, I'm torn between the two resolutions. I hate code duplication, but I also hate unnecessary code complication to avoid duplication. 

After a few greps, I'm biased towards the `save_esp()` modification. Why?` save_esp()` is s390 only and there pre-exist just three calls in `interp_masm_s390.cpp`. Plus, `save_esp()` is public anyway for some reason.